tests/test_fock.py

```diff
@@ -831,3 +831,143 @@ def circuit(n, a):
+                    (-rd1 ** 2 + rd1 ** 4 + (-0.5 + 2.5 * rd1 ** 2) * np.cosh(2 *\
+                    rs1) + 0.5 * np.cosh(4*rs1)) * np.sinh(4*rs0) +\
+                    rd1**2*np.cos(2 * phid1- phis1) * ((1 - 4 * rd0 ** 2) *\
+                    np.sinh(2 * rs0) - 1.5 * np.sinh(4 * rs0)) *\
+                    np.sinh(2 * rs1) + rd0**2*np.cos(2*phid0 - phis0) *\
+                    np.cosh( 2*rs0) * (-0.25 + 2 * rd1 ** 2 - 2 * rd1**4 + (1 -\
+                        4*rd1**2)* np.cosh(2*rs1) - 0.75*np.cosh(4* rs1) +\
+                        2*rd1**2*np.cos(2* phid1 - phis1) * np.sinh(2* rs1))
+
+        # differentiate wrt r of the first squeezing operation (rs0)
+        gradF = circuit.jacobian([pars], wrt={0}, method="F")
+        expected_gradient = pd_sr(*pars)
+        assert np.allclose(gradF, expected_gradient, atol=tol, rtol=0)
+
+        # differentiate wrt r of the second squeezing operation (rs1)
+        gradF = circuit.jacobian([pars], wrt={4}, method="F")
+
+        # can obtain the analytic expr for the partial derivate wrt r of the
+        # second squeezing operation by passing the parameters of operations
+        # acting on the second mode first
+        reverted_pars = np.concatenate([second_pars, first_pars])
+        expected_gradient = pd_sr(*reverted_pars)
+        assert np.allclose(gradF, expected_gradient, atol=tol, rtol=0)
+
+        def pd_dr(rs0, phis0, rd0, phid0, rs1, phis1, rd1, phid1):
+            """Analytic expression for the partial derivative with respect to
+            the r argument of the first displacement operation (rd0)"""
+            return rd0 * (0.5 - rd0**2 + (-2 + 4*rd0**2)*rd1**2*np.cosh( 2 * rs1)\
+                    + (-0.5 + rd0**2)*np.cosh(4 * rs1) + (2 - 4 * rd0**2)*\
+                    rd1**2 * np.cos( 2 * phid1 - phis1) * np.sinh(2*rs1) +\
+                    np.cosh(2 * rs0)*(-0.25 - 2*rd1**2 + 2 * rd1**4 + (-1 + 6\
+                        * rd1**2)*np.cosh(2 * rs1) + 1.25*np.cosh(4 * rs1) - 4\
+                        * rd1**2*np.cos(2 * phid1 - phis1)*np.sinh(2*rs1))\
+                    + np.cos(2 *phid0 - phis0)* np.sinh( 2 * rs0)*(-0.25 +\
+                        2 * rd1**2 - 2 * rd1**4 + (1 - 4 * rd1**2)*np.cosh(2*\
+                            rs1) - 0.75*np.cosh(4 * rs1) + 2*rd1**2*np.cos(2*\
+                                phid1 - phis1)*np.sinh(2 * rs1)))
+
+        # differentiate with respect to r of the first displacement operation (rd0)
+        gradF = circuit.jacobian([pars], wrt={2}, method="F")
+        expected_gradient = pd_dr(*pars)
+        assert np.allclose(gradF, expected_gradient, atol=tol, rtol=0)
+
+        # differentiate with respect to r of the second squeezing operation (rd1)
+        gradF = circuit.jacobian([pars], wrt={6}, method="F")
+
+        expected_gradient = pd_dr(*reverted_pars)
+        assert np.allclose(gradF, expected_gradient, atol=tol, rtol=0)
```

Comment:
Nice test!

Reply:
I wonder if this should be split up into 3-4 separate tests, for better readability?

Reply:
That's a good point! Was also thinking about it :thinking: Going for that would include having a longer part of code included in each test cases.

The main parts of these test cases could be factored out into fixtures before a new `TestVarianceDisplacedSqueezed` class:
```python
dev_list = [
            qml.device("strawberryfields.fock", wires=2, cutoff_dim=10),
            qml.device("strawberryfields.gaussian", wires=2)
            ]

@pytest.fixture(scope="function")
def disp_sq_circuit(dev):

    @qml.qnode(dev)
    def circuit(pars):
        qml.Squeezing(pars[0], pars[1], wires=0)
        qml.Displacement(pars[2], pars[3], wires=0)
        qml.Squeezing(pars[4], pars[5], wires=1)
        qml.Displacement(pars[6], pars[7], wires=1)
        return qml.var(qml.TensorN(wires=[0, 1]))

    return circuit

@pytest.fixture(scope="function")
def pars():
    """Parameters for the displaced squeezed circuit"""

    # Parameters to operations on the first mode
    # Squeezing
    rs0, phis0 = 0.1, 0.1

    # Displacement
    rd0, phid0 = 0.316277, 0.32175055
    alpha0 = rd0 * np.exp(1j*phid0)
    first_pars = np.array([rs0, phis0, rd0, phid0])

    # Parameters to operations on the second mode
    # Squeezing
    rs1, phis1 = 0.1, 0.15

    # Displacement
    rd1, phid1 = 0.14142136, 0.78539816
    alpha1 = rd1 * np.exp(1j*phid1)
    second_pars = np.array([rs1, phis1, rd1, phid1])

    return np.concatenate([first_pars, second_pars])
```

And each test case would then use the `disp_sq_circuit` and `pars` fixtures.

This would slightly differ from having explicit parts of the code in each test case (and could potentially end up in `conftest.py`). Would this be something that we'd like to go for or would we like to keep the tests entirely explicit?

Reply:
I was more asking about whether it makes sense to have so many different `assert`s in one function rather than having separate functions for each thing that is being tested.

Personally, when I read tests, I read them individually, so I don't even really notice code duplication. Fixtures are ok, as long as a user can easily find them and copy/paste them into their code

Reply:
makes sense, thank you! Broke it up into separate tests